internal/driver/flamegraph.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package driver
+
+import (
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"math"
+	"net/http"
+	"strings"
+
+	"github.com/google/pprof/internal/graph"
+	"github.com/google/pprof/internal/report"
+)
+
+type treeNode struct {
+	Name      string      `json:"n"`
+	Cum       int64       `json:"v"`
+	CumFormat string      `json:"l"`
+	Percent   string      `json:"p"`
+	Children  []*treeNode `json:"c"`
+}
+
+// percentage computes the percentage of total of a value, and encodes
+// it as a string. At least two digits of precision are printed.
+func percentage(value, total int64) string {
+	var ratio float64
+	if total != 0 {
+		ratio = math.Abs(float64(value)/float64(total)) * 100
+	}
+	return fmt.Sprintf("%5.2f%%", ratio)
+}
+
+// flamegraph generates a web page containing a flamegraph.
+func (ui *webInterface) flamegraph(w http.ResponseWriter, req *http.Request) {
+	// Force the call tree so that the graph is a tree.
+	rpt, errList := ui.makeReport(w, req, []string{"svg"}, "call_tree", "true")
```

Comment:
Please update this to
```
        // Force the call tree so that the graph is a tree. Also do not trim the tree
        // so that the flame graph contains all functions.
        rpt, errList := ui.makeReport(w, req, []string{"svg"}, "call_tree", "true", "trim", "false")
```

The `GetDOT` call trims the tree by default, and for the flame graph I think it's reasonable to override that.

Reply:
Up to you. I prefer to have the flame graph contain everything, but that's not the default behavior for the directed graph, so I wasn't really sure what would the users prefer.

Reply:
The graph does it by default because displaying the full graph for large programs becomes messy. Flame graph appears to be compact and readable enough on some larger profiles I tried so I think trimming the tree would be more confusing that useful here.